Apps/MAPL_GridCompSpecs_ACGv3.py

```diff
@@ -426,108 +280,145 @@ def dataframe(reader, columns):
+    for option_name, option in filter(lambda t: isinstance(t[1], dict), value_options.items()):
+        m = fetch_mapping_function(option.get(MAPPING))
+        (first, *tail) = get_from_keys(option)
+        name = option.get(AS, first if has_as_flag(option) else option_name) 
+        values[name] = m(*get_from_values((first, *tail), values, options[ARGDICT]))
+    return values, []
+
+def get_mandatory_option_keys(options):
+    keys = []
+    for tname, toptions in options.items():
+        if tname in {SPEC_ALIASES, CONSTANTS}:
+            continue
+        for key, option in toptions.items():
+            if is_mandatory(option):
+                keys.append(key)
+    return keys
+
+def get_internal_name(spec):
+    if spec is None:
+        return None
+    alias = spec.get(ALIAS, EMPTY).strip()
+    return alias if alias else spec.get(SHORT_NAME, EMPTY).replace('*', EMPTY)
+
+def get_values(specs, options):
+    all_values = []
+    results = []
+    aliases = options[SPEC_ALIASES]
+    flat_specs = flatten_specs(specs)
+    for spec in flat_specs:
+        dealiased = dict((aliases.get(k, k), v) for k, v in spec.items())
+        internal_name = get_internal_name(dealiased)
+        spec_values, specs_not_found = digest_spec(dealiased, options[SPECIFICATIONS])
+        values, values_not_found = map_spec_values(spec_values, options)
+        # Because the internal name is used in declare and get_pointer, it is singled out here.
+        values[INTERNAL_NAME] = internal_name
+        all_values.append(values)
+        mandatory_keys = get_mandatory_option_keys(options)
+        missing_mandatory = set(mandatory_keys).difference(values.keys())
+        result = {SPEC: spec,
+                  DEALIASED: dealiased,
+                  SPECS_NOT_FOUND: specs_not_found,
+                  VALUES_NOT_FOUND: values_not_found,
+                  MISSING_MANDATORY: missing_mandatory}
+        results.append(result)
+    return all_values, results
+
+def flatten_specs(specs):
+    match specs:
+        case Sequence():
+            flat_specs = list(specs)
+        case dict():
+            flat_specs = reduce(lambda a, c: a + c, specs.values(), [])
+    return flat_specs
     
+def flatten_options(o):
+    flat = {}
+    for v in o.values():
+        flat.update(v)
+    return flat 
 
 ################################# EMIT_VALUES ##################################
-def emit_values(specs, args):
+def emit_values(specs, options):
 
-    if args.name:
-        component = args.name
-    else:
-        component = os.path.splitext(os.path.basename(args.input))[0]
-        component = component.replace('_Registry','')
-        component = component.replace('_StateSpecs','')
+    argdict = options[ARGDICT]
+    exit_code_ = ERROR
+
+    add_newline = lambda s: f"{s.rstrip()}{linesep}"
+
+    component, declare_pointers, get_pointers = (argdict.get(k) for k in ('name', 'declare_pointers', 'get-pointers'))
```

Comment:
Should be `get_pointers` instead

Reply:
Fixed all the command-line options to use `-` and `_`. After reviewing the conventions for Unix-based operating systems, it appears that for command-line options, both are used, and the only real consistency is the `getopt` and `getopt_long` conventions, and after the `-` and `--`, there seem to be different approaches.